### DISCUSSION TITLE
.NET Core 2.1 container images will be deleted from Docker Hub

### DISCUSSION THREAD

**Discussion by richlander** [dotnet/dotnet-docker#3693] [Announcements]
# .NET Core 2.1 container images will be deleted from Docker Hub

Starting on August 21st, .NET Core 2.1 Docker container images will no longer be available on Docker Hub, but exclusively on Microsoft Container Registry (MCR). This change was previously announced with https://github.com/dotnet/dotnet-docker/issues/2848. If you are reliant on .NET Core 2.1 images on Docker Hub, you should switch to using MCR immediately. Please reach out at dotnet@microsoft.com if this change is a problem for you / your organization.

.NET Blog: https://devblogs.microsoft.com/dotnet/net-core-2-1-container-images-will-be-deleted-from-docker-hub/

## Details

[.NET images](https://hub.docker.com/_/microsoft-dotnet) are available on MCR from the following repos:

- `mcr.microsoft.com/dotnet/runtime-deps`
- `mcr.microsoft.com/dotnet/runtime`
- `mcr.microsoft.com/dotnet/aspnet`
- `mcr.microsoft.com/dotnet/sdk`

You need to make the following transformations to pull .NET Core 2.1 images from MCR instead of Docker Hub.

For the SDK:

```Dockerfile
microsoft/dotnet:2.1-sdk -> mcr.microsoft.com/dotnet/sdk:2.1
microsoft/dotnet:2-sdk -> mcr.microsoft.com/dotnet/sdk:2.1
microsoft/dotnet:2.1-sdk-stretch -> mcr.microsoft.com/dotnet/sdk:2.1-stretch
microsoft/dotnet:2.1-sdk-stretch-arm32v7 -> mcr.microsoft.com/dotnet/sdk:2.1-stretch-arm32v7
microsoft/dotnet:2.1-sdk-nanoserver-1809 -> mcr.microsoft.com/dotnet/sdk:2.1-nanoserver-1809
microsoft/dotnet:2.1-sdk-alpine -> mcr.microsoft.com/dotnet/sdk:2.1-alpine
microsoft/dotnet:2.1-sdk-bionic -> mcr.microsoft.com/dotnet/sdk:2.1-bionic
microsoft/dotnet:2.1-sdk-bionic-arm32v7 -> mcr.microsoft.com/dotnet/sdk:2.1-bionic-arm32v7
microsoft/dotnet:latest -> mcr.microsoft.com/dotnet/sdk:2.1
```

For the ASP.NET Core:

```Dockerfile
microsoft/dotnet:2.1-aspnetcore-runtime -> mcr.microsoft.com/dotnet/aspnet:2.1
microsoft/dotnet:2-aspnetcore-runtime-> mcr.microsoft.com/dotnet/aspnet:2.1
microsoft/dotnet:2.1-aspnetcore-runtime-stretch-slim -> mcr.microsoft.com/dotnet/aspnet:2.1-stretch-slim
microsoft/dotnet:2.1-aspnetcore-runtime-stretch-slim-arm32v7 -> mcr.microsoft.com/dotnet/aspnet:2.1-stretch-slim-arm32v7
microsoft/dotnet:2.1-aspnetcore-runtime-nanoserver-1809 -> mcr.microsoft.com/dotnet/aspnet:2.1-nanoserver-1809
microsoft/dotnet:2.1-aspnetcore-runtime-alpine -> mcr.microsoft.com/dotnet/aspnet:2.1-alpine
microsoft/dotnet:2.1-aspnetcore-runtime-bionic -> mcr.microsoft.com/dotnet/aspnet:2.1-bionic
microsoft/dotnet:2.1-aspnetcore-runtime-bionic-arm32v7 -> mcr.microsoft.com/dotnet/aspnet:2.1-bionic-arm32v7
microsoft/dotnet:aspnetcore-runtime-> mcr.microsoft.com/dotnet/aspnet:2.1
```

For .NET Runtime:

```Dockerfile
microsoft/dotnet:2.1-runtime -> mcr.microsoft.com/dotnet/runtime:2.1
microsoft/dotnet:2-runtime -> mcr.microsoft.com/dotnet/runtime:2.1
microsoft/dotnet:2.1-runtime-stretch-slim -> mcr.microsoft.com/dotnet/runtime:2.1-stretch-slim
microsoft/dotnet:2.1-runtime-stretch-slim-arm32v7 -> mcr.microsoft.com/dotnet/runtime:2.1-stretch-slim-arm32v7
microsoft/dotnet:2.1-runtime-nanoserver-1809 -> mcr.microsoft.com/dotnet/runtime:2.1-nanoserver-1809
microsoft/dotnet:2.1-runtime-alpine -> mcr.microsoft.com/dotnet/runtime:2.1-alpine
microsoft/dotnet:2.1-runtime-bionic -> mcr.microsoft.com/dotnet/runtime:2.1-bionic
microsoft/dotnet:2.1-runtime-bionic-arm32v7 -> mcr.microsoft.com/dotnet/runtime:2.1-bionic-arm32v7
microsoft/dotnet:runtime -> mcr.microsoft.com/dotnet/runtime:2.1
```

For .NET Runtime dependencies:

```Dockerfile
microsoft/dotnet:2.1-runtime-deps -> mcr.microsoft.com/dotnet/runtime-deps:2.1
microsoft/dotnet:2.1-runtime-deps-stretch-slim -> mcr.microsoft.com/dotnet/runtime-deps:2.1-stretch-slim
microsoft/dotnet:2.1-runtime-deps-stretch-slim-arm32v7 -> mcr.microsoft.com/dotnet/runtime-deps:2.1-stretch-slim-arm32v7
microsoft/dotnet:2.1-runtime-deps-alpine -> mcr.microsoft.com/dotnet/runtime-deps:2.1-alpine
microsoft/dotnet:2.1-runtime-deps-bionic -> mcr.microsoft.com/dotnet/runtime-deps:2.1-bionic
microsoft/dotnet:2.1-runtime-deps-bionic-arm32v7 -> mcr.microsoft.com/dotnet/runtime-deps:2.1-bionic-arm32v7
microsoft/dotnet:runtime-deps -> mcr.microsoft.com/dotnet/runtime-deps:2.1
```

**Comment by ghost** [comment]
OK， The day come. we have upgrade to dotnet 3.1 LTS. and waiting for next LTS .net.
Thanks for great works anyway.

**Comment by richlander** [comment]
Excellent. You'll be able to use .NET 6 (which is LTS) in production starting with RC1, which we'll release in the first half of next month. We encourage teams to do that to get the new value sooner. We're happy to help with your .NET 6 adoption. In fact, I have a customer call on that topic later today.

**Comment by jmartins-sh** [comment]
Hey guys, how are you going? 

How about SDK and Runtime from 2.2? 

Yesterday I tried to build an image using SDK 2.2 (mcr.microsoft.com/dotnet/sdk:2.2) but I got a not found error. 😿  

Kind regards! 😸

**Comment by mthalman** [comment]
> Yesterday I tried to build an image using SDK 2.2 (mcr.microsoft.com/dotnet/sdk:2.2) but I got a not found error. 😿

That's because .NET Core 2.2 is no longer supported.  It has been [out of support](https://github.com/dotnet/dotnet-docker/issues/1542) since Dec 2019.  Note that .NET Core 2.2 was not an LTS release.  It is recommended that you upgrade to the latest supported version: .NET 5.0.

**Comment by jmartins-sh** [comment]
> > Yesterday I tried to build an image using SDK 2.2 (mcr.microsoft.com/dotnet/sdk:2.2) but I got a not found error. 😿
> 
> That's because .NET Core 2.2 is no longer supported. It has been [out of support](https://github.com/dotnet/dotnet-docker/issues/1542) since Dec 2019. Note that .NET Core 2.2 was not an LTS release. It is recommended that you upgrade to the latest supported version: .NET 5.0.

Awesome @mthalman, thanks for the reply! Well, I guess I gotta update our apps to the LTS version! \o/ 

Have a nice one! 😸

**Comment by MichaelSimons** [comment]
@joaoantoniomartinsfilho, If needed you can still pull the 2.2 images - mcr.microsoft.com/dotnet/core/sdk:2.2.  There was a repo rename during the 5.0 release and only the in support releases were copied to the new repos. See https://github.com/dotnet/dotnet-docker/issues/1939 for details.

But as @mthalman mentioned, you should update to a supported version of .NET.
